src/ensembl/src/content/app/entity-viewer/gene-view/components/gene-view-sidebar/overview/GeneOverview.tsx

```diff
@@ -15,119 +15,78 @@
  */
 
 import React from 'react';
-import { connect } from 'react-redux';
+import { useParams } from 'react-router-dom';
+import { useQuery, gql } from '@apollo/client';
 
 import ExternalReference from 'src/shared/components/external-reference/ExternalReference';
-import MainAccordion from './MainAccordion';
 
-import {
-  getEntityViewerSidebarPayload,
-  getEntityViewerSidebarUIState
-} from 'src/content/app/entity-viewer/state/sidebar/entityViewerSidebarSelectors';
-import { updateEntityUI } from 'src/content/app/entity-viewer/state/sidebar/entityViewerSidebarActions';
+import { parseEnsObjectIdFromUrl } from 'src/shared/state/ens-object/ensObjectHelpers';
 
-import { RootState } from 'src/store';
-import {
-  EntityViewerSidebarPayload,
-  EntityViewerSidebarUIState
-} from 'src/content/app/entity-viewer/state/sidebar/entityViewerSidebarState';
+import { EntityViewerParams } from 'src/content/app/entity-viewer/EntityViewer';
+import { Gene } from 'src/content/app/entity-viewer/types/gene';
 
 import styles from './GeneOverview.scss';
 
-type Props = {
-  sidebarPayload: EntityViewerSidebarPayload | null;
-  sidebarUIState: EntityViewerSidebarUIState | null;
-  updateEntityUI: (uIstate: Partial<EntityViewerSidebarUIState>) => void;
-};
+const QUERY = gql`
+  query Gene($genomeId: String!, $geneId: String!) {
+    gene(byId: { genome_id: $genomeId, stable_id: $geneId }) {
+      alternative_symbols
+      name
+      stable_id
+      symbol
+    }
+  }
+`;
+
+const GeneOverview = () => {
+  const params: EntityViewerParams = useParams();
+  const { entityId, genomeId } = params;
+  const stableId = entityId ? parseEnsObjectIdFromUrl(entityId).objectId : null;
+
+  const { data, loading } = useQuery<{ gene: Gene }>(QUERY, {
+    variables: {
+      stable_id: stableId,
+      genome_id: genomeId
+    },
+    skip: !stableId
+  });
 
-const GeneOverview = (props: Props) => {
-  if (!props.sidebarPayload) {
+  if (loading) {
+    return <div>Loading...</div>;
+  }
+
+  if (!data || !data.gene) {
     return <div>No data to display</div>;
   }
-  const { gene } = props.sidebarPayload;
+
+  const { gene } = data;
+
   return (
     <div className={styles.overviewContainer}>
       <div className={styles.geneDetails}>
         <span className={styles.geneSymbol}>{gene.symbol}</span>
-        <span>{gene.id}</span>
+        <span>{gene.stable_id}</span>
       </div>
 
       <div className={styles.sectionHead}>Gene name</div>
 
       <div className={styles.geneName}>
         <ExternalReference
-          label={gene.metadata?.name?.description}
-          linkText={gene.metadata?.name?.source?.id || ''}
+          label={gene.name as string}
```

Comment:
But gene.name can really be null, typescript isn't lying.

Reply:
I've removed `ExternalReference` to resolve a comment below.

Reply:
✅